peribolos-fix/Dockerfile

```diff
@@ -0,0 +1,12 @@
+FROM golang:latest
+RUN apt update && apt install -y patch
+WORKDIR /peribolos
+# As of now the last commit, not downloading the latest becuase the patch might not work
+RUN curl -L https://github.com/kubernetes/test-infra/tarball/master/83bf07dd99d23a5d6998490ec35240a4be895254 | tar -xz
+WORKDIR /peribolos/kubernetes-test-infra-83bf07d/prow/cmd/peribolos
+# Patch file encoded in base64
+RUN echo "LS0tIG1haW4uZ28JMjAyMi0wNC0xMyAxOTozOTo1NS41NTMxMzYzNDYgKzAyMDAKKysrIG1haW4tY2hhbmdlLmdvCTIwMjItMDQtMTMgMTk6MDI6NDIuNzU4MzEzNDgzICswMjAwCkBAIC0yNDEsMTEgKzI0MSw2IEBACiAJb3V0Lk1ldGFkYXRhLkRlZmF1bHRSZXBvc2l0b3J5UGVybWlzc2lvbiA9ICZkcnAKIAlvdXQuTWV0YWRhdGEuTWVtYmVyc0NhbkNyZWF0ZVJlcG9zaXRvcmllcyA9ICZtZXRhLk1lbWJlcnNDYW5DcmVhdGVSZXBvc2l0b3JpZXMKIAotCXZhciBydW5uaW5nQXNBZG1pbiBib29sCi0JcnVubmluZ0FzLCBlcnIgOj0gY2xpZW50LkJvdFVzZXIoKQotCWlmIGVyciAhPSBuaWwgewotCQlyZXR1cm4gbmlsLCBmbXQuRXJyb3JmKCJmYWlsZWQgdG8gb2J0YWluIHVzZXJuYW1lIGZvciB0aGlzIHRva2VuIikKLQl9CiAJYWRtaW5zLCBlcnIgOj0gY2xpZW50Lkxpc3RPcmdNZW1iZXJzKG9yZ05hbWUsIGdpdGh1Yi5Sb2xlQWRtaW4pCiAJaWYgZXJyICE9IG5pbCB7CiAJCXJldHVybiBuaWwsIGZtdC5FcnJvcmYoImZhaWxlZCB0byBsaXN0IG9yZyBhZG1pbnM6ICV3IiwgZXJyKQpAQCAtMjU0LDEzICsyNDksNiBAQAogCWZvciBfLCBtIDo9IHJhbmdlIGFkbWlucyB7CiAJCWxvZ3J1cy5XaXRoRmllbGQoImxvZ2luIiwgbS5Mb2dpbikuRGVidWcoIlJlY29yZGluZyBhZG1pbi4iKQogCQlvdXQuQWRtaW5zID0gYXBwZW5kKG91dC5BZG1pbnMsIG0uTG9naW4pCi0JCWlmIHJ1bm5pbmdBcy5Mb2dpbiA9PSBtLkxvZ2luIHsKLQkJCXJ1bm5pbmdBc0FkbWluID0gdHJ1ZQotCQl9Ci0JfQotCi0JaWYgIXJ1bm5pbmdBc0FkbWluIHsKLQkJcmV0dXJuIG5pbCwgZm10LkVycm9yZigiLS1kdW1wIG11c3QgYmUgcnVuIHdpdGggYWRtaW46b3JnIHNjb3BlIHRva2VuIikKIAl9CiAKIAlvcmdNZW1iZXJzLCBlcnIgOj0gY2xpZW50Lkxpc3RPcmdNZW1iZXJzKG9yZ05hbWUsIGdpdGh1Yi5Sb2xlTWVtYmVyKQo="\
+| base64 -d > peribolos.patch
+RUN patch -f -u main.go -i peribolos.patch
+RUN go build && mv peribolos /usr/bin/peribolos
+CMD /bin/bash
```

Comment:
I think this image is unnecessarily large. You can use a multi stage build and in the first stage you build the binary, in the second stage you just copy it over and use it as an entry point. I don't think we need anything else in the image than the peribolos binary.

https://chemidy.medium.com/create-the-smallest-and-secured-golang-docker-image-based-on-scratch-4752223b7324

Reply:
Thanks for the reference!

Reply:
Since I couldn't find a way to use the `peribolos` binary as an entrypoint, I used the `apline` image to use with `sh`. I can change it if we find a way to just use the binary.